src/FakeItEasy/Configuration/FakeConfigurationManager.cs

```diff
@@ -61,11 +62,14 @@ public IAnyCallConfigurationWithNoReturnTypeSpecified CallTo(object fakeObject)
         public IPropertySetterAnyValueConfiguration<TValue> CallToSet<TValue>(Expression<Func<TValue>> propertySpecification)
         {
             Guard.AgainstNull(propertySpecification, nameof(propertySpecification));
-            this.AssertThatMemberCanBeIntercepted(propertySpecification);
+            var parsedCallExpression = this.callExpressionParser.Parse(propertySpecification);
+            this.AssertThatMemberCanBeIntercepted(parsedCallExpression);
 
-            var setterExpression = BuildSetterFromGetter(propertySpecification);
+            var setterExpression = BuildSetterFromGetter(parsedCallExpression, propertySpecification);
 
-            return new PropertySetterConfiguration<TValue>(setterExpression, this.CreateVoidArgumentValidationConfiguration);
+            return new PropertySetterConfiguration<TValue>(
+                setterExpression,
+                lambda => this.CreateVoidArgumentValidationConfiguration(this.callExpressionParser.Parse(lambda)));

```

Comment:
It's not the most obvious, it's true. An alternative would be to pass the parser into the `PropertySetterConfiguration`.
**also**, I had not intended to call that value `lambda`. :blush:. Something like `propertySetterExpression` would be better, I think.